File: src/main/java/com/codingforcookies/betterrecords/api/wire/IRecordWireHome.java
package com.codingforcookies.betterrecords.api.wire;

import java.util.ArrayList;

import com.codingforcookies.betterrecords.api.connection.RecordConnection;
import net.minecraft.item.ItemStack;
import net.minecraft.tileentity.TileEntity;

public interface IRecordWireHome {
    void addTreble(float form);
    void addBass(float form);

    ArrayList<RecordConnection> getConnections();
    void increaseAmount(IRecordWire wireComponent);
    void decreaseAmount(IRecordWire wireComponent);
    float getSongRadius();
    TileEntity getTileEntity();
    ItemStack getItemStack();
}

File: src/main/kotlin/com/codingforcookies/betterrecords/api/wire/IRecordWireHome.kt
package com.codingforcookies.betterrecords.api.wire

import com.codingforcookies.betterrecords.api.connection.RecordConnection
import net.minecraft.item.ItemStack
import net.minecraft.tileentity.TileEntity

interface  IRecordWireHome {

    fun addTreble(form: Float)
    fun addBass(form: Float)

    val connections: MutableList<RecordConnection>

    fun increaseAmount(wireComponent: IRecordWire)
    fun decreaseAmount(wireComponent: IRecordWire)

    val songRadius: Float

    val tileEntity: TileEntity
    val record: ItemStack

    var wireSystemInfo: HashMap<String, Int>
}
